payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncher.kt

```diff
@@ -35,12 +54,58 @@ class GooglePayPaymentMethodLauncher @AssistedInject internal constructor(
     @Assisted private val config: Config,
     @Assisted private val readyCallback: ReadyCallback,
     @Assisted private val activityResultLauncher: ActivityResultLauncher<GooglePayPaymentMethodLauncherContract.Args>,
```

Comment:
(Optional and not needed in this PR) - instead of a activityResultLauncher, I think we can take a `ActivityResultCaller` and 
`ActivityResultCallback<GooglePayPaymentMethodLauncher.Result>` here and register it within, this way `GooglePayPaymentMethodLauncherContract` is hidden from its callers

Reply:
`GooglePayPaymentMethodLauncherContract` still needs to be public so that `GooglePayPaymentMethodLauncher` can be used from Compose.

Reply:
Yeah I think we should phase it out by creating a public factory method instead as `GooglePayPaymentMethodLauncherContract` exposes impl details such as injector keys

Reply:
The `InjectionParams` is internal, and the constructor that takes it is internal too. Agree with having a factory, it's a pattern that we should follow everywhere to give us more flexibility over the constructors.